### DISCUSSION TITLE
ipv6 proxy

### DISCUSSION THREAD

**Discussion by Ch4ttY** [thedemons/opentele#52] [General]
I don't have ipv6 SOCKS5 proxies working with your library, does your library support this? Am I doing something wrong?

import socks
from opentele.tl import TelegramClient

tdesk = TDesktop(account)
            api = API.TelegramDesktop.Generate()

            client = await TelegramClient.FromTDesktop(tdesk,
                                                       flag=UseCurrentSession,
                                                       api=api,
                                                       use_ipv6=True,
                                                       proxy=(socks.SOCKS5,
                                                              '217.22.62.241',
                                                              10111,
                                                              True,
                                                              'HzJ9PZ',
                                                              'ngEXQ3')
                                                       )

            await client.connect()

**Comment by thedemons** [comment]
> I don't have ipv6 SOCKS5 proxies working with your library, does your library support this? Am I doing something wrong?
> 
> import socks
> from opentele.tl import TelegramClient
> 
> tdesk = TDesktop(account)
>             api = API.TelegramDesktop.Generate()
> 
>             client = await TelegramClient.FromTDesktop(tdesk,
>                                                        flag=UseCurrentSession,
>                                                        api=api,
>                                                        use_ipv6=True,
>                                                        proxy=(socks.SOCKS5,
>                                                               '217.22.62.241',
>                                                               10111,
>                                                               True,
>                                                               'HzJ9PZ',
>                                                               'ngEXQ3')
>                                                        )
> 
>             await client.connect()

The TelegramClient class in `opentele.tl` is just a wrapper for telethon's TelegramClient, it basicly just pass the arguments to telethon. I've never use ipv6 proxy before and I didn't test it yet but the code above should work just fine.

Just to confirm, this is working for telethon but not for opentele, right?

I'll look into it and find out what went wrong.

**Comment by Ch4ttY** [comment]
It worked like that, I have deleted 'use_ipv6=True'. 

client = await TelegramClient.FromTDesktop(tdesk,
                                                   flag=UseCurrentSession,
                                                   api=api,
                                                   proxy=(socks.SOCKS5,
                                                          '217.22.62.241',
                                                          10111,
                                                          True,
                                                          'HzJ9PZ',
                                                          'ngEXQ3')
                                                   )
                                                   
As far as I know, Telethon has stopped supporting ipv6, but opentele supports it, is the old version being used?

**Comment by thedemons** [comment]
> It worked like that, I have deleted 'use_ipv6=True'. 
> 
> client = await TelegramClient.FromTDesktop(tdesk,
>                                                    flag=UseCurrentSession,
>                                                    api=api,
>                                                    proxy=(socks.SOCKS5,
>                                                           '217.22.62.241',
>                                                           10111,
>                                                           True,
>                                                           'HzJ9PZ',
>                                                           'ngEXQ3')
>                                                    )
>                                                    
> As far as I know, Telethon has stopped supporting ipv6, but opentele supports it, is the old version being used?

Opentele uses the installed version of telethon on your computer, it doesn't include or install any other version, if telethon stopped supporting ipv6 then also opentele.

You could try and installed older version of telethon that supports ipv6
